src/seedu/addressbook/Main.java

```diff
@@ -103,27 +115,58 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) {
         try {
+            checkIfFileExists(storage.getFile());
```

Comment:
`checkIfFileExists(...)` seems to be at a lower abstraction level then the other methods here.
It would be better to for this to be contained inside `storage.save(..)` instead as `StorageFile` seems to be more qualified to handle file exceptions.